polyhook2/PE/EatHook.hpp

```diff
@@ -31,10 +33,15 @@ class EatHook : public IHook {
 	virtual HookType getType() const override {
 		return HookType::EAT;
 	}
-private:
+
+protected:

```

Comment:
why the change to protected? I wouldn't expect this to ever be a parent class really, do you see a case someone may want to derive from this as a base class?

Reply:
In general, I consider it a sub-optimal design choice to set any class members to private for library classes, because if there is going to be a need to modify the behavior for unforeseeable reasons, the consumer of the library won't be cursing the library author(s) for needlessly restricting access. It's not a pleasant experience to maintain a private fork of the library because it is impossible to inherit a library class to modify it. But that's just my opinion, and I wanted to hear your view on this issue as well.

Reply:
That's reasonable justification honestly. I set classes I expected to be derived from to protected already, I just never expected a case it would be useful for this but it doesn't hurt so why not.